src/isomorphic/classic/class/ReactClass.js

```diff
@@ -437,6 +437,19 @@ function validateMethodOverride(isAlreadyDefined, name) {
  */
 function mixSpecIntoComponent(Constructor, spec) {
   if (!spec) {
+    if(__DEV__) {
+      var typeofSpec = typeof spec;
+
+      warning(
+        typeofSpec !== 'object' || typeofSpec === null,
```

Comment:
Also, I think the check here might be backwards.  Warning behaves the opposite of an if statement, if I recall correctly.  We should probably add a couple of unit tests to verify behavior.